src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -690,7 +691,13 @@ private CallResult ExecutePrecompile(EvmState state, IReleaseSpec spec)
         try
         {
             (ReadOnlyMemory<byte> output, bool success) = precompile.Run(callData, spec);
-            CallResult callResult = new(output.ToArray(), success, !success);
+            byte[] result = (MemoryMarshal.TryGetArray(output, out ArraySegment<byte> outputArray) &&
+                outputArray.Offset == 0 && outputArray.Count == output.Length) ?
+                outputArray.Array :
+                // Wasn't array backed or Memory wasn't the whole array
+                output.ToArray();
+
+            CallResult callResult = new(result, success, !success);
```

Comment:
We can just change `CallResult` to accept `ReadOnlyMemory<byte>` as it is already like this in property:

```
public ReadOnlyMemory<byte> Output { get; }
```

Reply:
Changed